pkg/handler/handler.go

```diff
@@ -24,7 +26,7 @@ import (
 // processed on demand would fit into this category. For instance, Branch ENIs are tied to the Security
 // Group required by the pod which we would know only after receiving the pod request.
 type Handler interface {
-	CanHandle(resourceName string) bool
-	HandleCreate(resourceName string, requestCount int, pod *v1.Pod) (ctrl.Result, error)
-	HandleDelete(resourceName string, pod *v1.Pod) (ctrl.Result, error)
+	GetProvider() provider.ResourceProvider
```

Comment:
With the terminology and the design here, I have some questions.

- Usually a Handler exposes the methods for handling operations (like created, delete, update) here.
- Adding `GetProvider()` to Handler interface, seems to me to add two different functions in the interface.

I understood the behavior change, this is a common the code organization or interface design.

Reply:
Agreed, will remove the GetProvider() from the interface.